devices/debian_wifi.py

```diff
@@ -1,88 +1,109 @@
-import re
+import re, time
 import debian
 import pexpect
+from countrycode import countrycode
+from lib.wifi import wifi_client_stub
 
-class DebianWifi(debian.DebianBox):
+class DebianWifi(debian.DebianBox, wifi_client_stub):
     '''Extension of Debian class with wifi functions'''
 
     model = ('debianwifi')
+    iface_wlan = "wlan0"
+    iface_wlan1 = "wlan1"
 
-    iface_wlan = "wlan1"
+    def disable_and_enable_wifi(self, iface):
+       self.set_link_state(iface,"down")
+       self.set_link_state(iface,"up")
 
-    def scan(self):
+    def disable_wifi(self, iface):
+       self.set_link_state(iface,"down")
+
+    def enable_wifi(self, iface):
+       self.set_link_state(iface,"up")
+
+    def wifi_scan(self, iface):
         from tests.lib.installers import install_iw
         install_iw(self)
 
-        self.sendline('iw %s scan | grep SSID:' % self.iface_wlan)
-        self.expect(self.prompt)
+        output = self.sudo_sendline('iw %s scan | grep SSID:' % iface)
```

Comment:
If you want the output, you should do an expect.

sendline() never returns matched chars, just the number of bytes sent.